bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/tools/ToolsActivationJob.java

```diff
@@ -189,22 +190,16 @@ private void setEnvVarsInEclipse()
 	{
 		if (idfToolSet == null)
 			throw new RuntimeException("Tools Cannot be null");
-		
+
 		IDFEnvironmentVariables idfEnvironmentVariables = new IDFEnvironmentVariables();
 		idfEnvironmentVariables.removeAllEnvVariables();
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.ESP_IDF_VERSION, idfToolSet.getIdfVersion());
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.GIT_PATH, idfToolSet.getSystemGitExecutablePath());
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.IDF_COMPONENT_MANAGER, idfToolSet.getEnvVars().get(IDFEnvironmentVariables.IDF_COMPONENT_MANAGER));
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.IDF_MAINTAINER, idfToolSet.getEnvVars().get(IDFEnvironmentVariables.IDF_MAINTAINER));
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.IDF_PATH, idfToolSet.getIdfLocation());
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.IDF_PYTHON_ENV_PATH, idfToolSet.getEnvVars().get(IDFEnvironmentVariables.IDF_PYTHON_ENV_PATH));
-		idfEnvironmentVariables.addEnvVariable(IDFEnvironmentVariables.OPENOCD_SCRIPTS, idfToolSet.getEnvVars().get(IDFEnvironmentVariables.OPENOCD_SCRIPTS));
-		
+		idfToolSet.getEnvVars().forEach(idfEnvironmentVariables::addEnvVariable);
```

Comment:
LGTM just please see if this is NULL safe as at  times the json in the idfToolSet for env vars can contain nulls